src/main/java/com/iwaconsolti/league/controller/controller/MatchController.java

```diff
@@ -0,0 +1,39 @@
+package com.iwaconsolti.league.controller.controller;
+import com.iwaconsolti.league.model.MatchModel;
+import com.iwaconsolti.league.service.MatchService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+
+
+@RestController
+@RequestMapping("/Match")
+@Slf4j
+public class MatchController {
+
+    private final MatchService service;
+    @Autowired
+    public MatchController(MatchService service) {
+
+        this.service = service;
+    }
+
+    @GetMapping("/team/{id}")
+    public MatchModel getMatchById(@PathVariable int id) {
+        return service.getMatchById(id);
+    }
+
+    @PostMapping("/")
+    public MatchModel addMatch (@RequestBody MatchModel matchmodel) {
+
+        return service.insertMatch(matchmodel );
+    }
+
+    @DeleteMapping("/matches")
```

Comment:
deleteMatch should accept an ID parameter or be more specific if it is a bulk operation.